### DISCUSSION TITLE
How to get req.query req.params if we dont know the shape

### DISCUSSION THREAD

**Discussion by fbritoferreira** [RobinTail/express-zod-api#831] [Q&A] [answered]
I have an app where we want to accept a wide number of params and query and pull it all together as a object.

I was wandering if i we could use `input: z.record(z.string(), z.string())` it would accept them and pull those together

Existing Middleware
```js

       const options = {};

        Object.keys(req.query).forEach(param => options[param] = req.query[param]);
        Object.keys(req.params).forEach(param => options[param] = req.params[param]);

        req.opts = options;

```

**Comment by RobinTail** [comment, marked as answer]
Hello @fbritoferreira ,

`request` properties are combined and into a single object that is being validated against input schema, which has to be `z.object`.
However, you can easily enable unknown properties on that schema by utilizing `.passthrough()` method.

Check out this example:

https://github.com/RobinTail/express-zod-api/blob/master/example/endpoints/upload-avatar.ts#L18

Another advice could be making your `z.record` schema a property inside the input schema object, like this:
```ts
z.object({ 
  params: z.record(
    z.string(), z.string(),
  )
})
```

I hope this would help you

**Reply by RobinTail** [reply]
I forget to mention, that `.passthrough()` on schema is only available for Endpoint.
If you need it for Middleware, then you solution with manual examination of `req.body` and `req.params` and putting them into `options` might be the best one.

However, I'd still recommend using a property inside of `options` for all your unknown inputs, @fbritoferreira
